### ISSUE TITLE
Add time-base window to negate incoming garbage line and/or decrease combo Quad damage

### ISSUE THREAD

**Issue by bmjeon5957** [tetrio/issues#435]
### Checklist (issue will be disregarded & closed if incomplete!)

- [x] I've **checked both [open and recently closed issues](https://github.com/tetrio/issues/issues?q=is%3Aissue)** to ensure this feature is not already requested.
- [x] I am using the **newest version** of TETR.IO.
- [x] I am certain this is a **feature request**, not a bug.
- [x] I've **thought this feature through** properly.

-----

### Is your feature request related to a problem? Please describe.

1. Players are forced to play in a very downstack-heavy playstyle, especially on high(SS+) level gameplay. 
    Setting up various T-spin setups, or even aiming for Quad is very discouraged except as a opener or early game aggression.

 Reference : basically any X/U gameplay youtube.com "tetrio" 
 Reference 2 : any competetive Puyo Puyo tetris gameplay that does not include 4-wide (recommending kazu vs zetris)

2. Game-finishing attacks, aka a Quad/T-spin combo, is mostly a result of RNG, and is less related to skill.
  This issue is closely tied to (1)'s issue. Because you realistically cannot negate incoming damage promptly, you're forced to maintain a low playfield to not lose in a frustrating way. This discourages "setting up" for your back-to-backs as complicated tspins require higher playfield.

 This leads to players not being able to setup high damage attacks by themselves, instead rely on garbage line which seems to have very forgiving algorithm, making Quad very easy. However combing into Quad is another issue, as it requires I piece at the right time, when you have the right playfield. 

Worthy mentioning that i'm using the word "realistically", as in theory you do you have way to counter incoming damage, However in practice it's mostly a result of coincidence or at most , player opting for doubles instead of finishing their own tspin/quad setup on very high stacks.

  
### Describe the solution you'd like

1. Add a time based window to counter incoming garbage lines. Time i suggest is 2-3 seconds, preferbly 3s, as It allows players playing in 2PPS to cycle through one bag with hold's help.

There are various ways of implementing this, as Tetris99 (attacks change from gray to red) and Puyo Puyo tetris (Long line clear delay + damages can be deflected while "flying").

I have no clear idea on how damage incoming is implemented in tetrio, as in if each attacks are treated as different instances or is basically add of numbers, but having a another buffer (it's pretty much already implemented as in : attack flying time) before attack actually being added to the real damage bar, and being able to counter it while it's flying seems clear and implementable enough.

2. Preferbly reduce the damage bonus for quad, However, if (1) is implemented this might not be required.

### Describe alternatives you've considered

1. Quad combo damage needs to be toned down even moreso in case if time-based window to negate damage is not added. see : frustration the RNGness.

2. Alternatively, garbage lines could generally be much more "messy" making most of game-ending quad combo much rarer. This is actually also a OK solution to "people are encouraged to not setup their own attack" playstyle, however will lead every game to end very prematurely compared to now.

3. Although things won't be too different if players have skill difference, yes this will make games longer especially between similarly skilled players. After a testing period, it might be OK to reduce FT 3/5/7 to something like 2/3/5 respectively, to keep average playtime per one ranked queue remains similar.

### Additional context

It's a small community but I'm seeing reasonable amounts of post/replys in korean stacker community that states this as a major reason as in why they're returning to the (game they were originally playing). I want to be clear that I'm not really trying to be rude/indirectly threaten you, just thinking that it's worth mentioning.

I've also made similar requests, especially (1), thrice for now.  It has gained major positive response in #feature_request , as shown in reactions. None of the requests near mines had close amount of so called "agreed" reactions. Yes not all the players see the channel. However it feels pretty safe to assume that people wants this fixed when a request is getting 10+ positive reactions opposed to most of requests getting no attention.

Making players feel like "I don't deserve this death, i have 3 tspins available for me" or "This quad combo burst is almost a guarrenteed kill but it's pretty much RNG" combined with somewhat laggy-nature of enemy field/attack(I believe this is inevitable) make players very frustrated and often quit. Funny enough, we also see a lot of ragequits.

It's fine when you're overwhelmed with attacks by a better player, even if you've done everything you can do. But the sad fact is that at current state people are getting many 'undeserving deaths' in forms of dying after making two tspins and having a tetris well open, because of the how game currently works.

I highly suggest to have this fixed, and many people agrees. 

I don't exactly want to be rude,  But I haven't been given the answer. If this is opposed to your philosophy, like that's fine. If you just don't feel like it, that's also fine. I'm not asking for rocket science reasons, nor insisting/blaming on you. People including me is basically free to walk away and we're not forced, I understand.

But I'm having trouble understanding on why we're getting no response on feature that's gaining fairly big amount of agreement, and is probably one of the biggest reason as in if people would keep playing this game or not.

Can we get an answer?

**Comment by o5k** [comment]
This phenomenon is called "grazing". It happens when either...

1. The recipient sends garbage before even knowing garbage is coming towards them (caused by lag / high ping)
2. The recipient sends garbage while garbage is flying towards them. (A [special sound](https://kagari.moe/:/exchange.wav) plays when this occurs.)

Case 1 cannot be fixed, simply due to how the game's netcode works. In games like _Puyo Puyo Tetris_, this is fixed, but this in turn causes the entire game to freeze while someone is lagging. Not a good way to work.

Case 2 is totally fixable, however I don't want to make such a change just yet. You see, fixing case 2 essentially **rolls back something that makes TETR.IO different**. It makes TETR.IO act more like other Guideline games (especially if I also added garbage delay before it enters your field). That is **not** a move I want to make right now - I first want to see how players adapt to the current mechanics.

It's possible that even after another many months, people will still feel it plays like RNG, and the new mechanics are simply not as good as the more Guideline ones. However, that is something that I don't think can be said just yet, as there simply has not been enough experimentation yet. I don't want to abandon these mechanics just by virtue of some not being willing to try to learn the new mechanics.

Of course, I will follow this as the meta keeps changing. I've marked this issue to never go stale, so feel free to comment here when you think the situation has changed, or you simply have something to say.

**Comment by bmjeon5957** [comment]
If you think it needs more time, ok sure, but i feel like we'd have enough time and it's in sooner the better category. At least that's how i think.

Granted, the change does not have to be in form of adding time based counter window. Something that you'd like to keep should stay. But do consider that **some factors are basically multiplicatively generating frustrations for players, being significantly worse as skill level goes up**. and those factors are :

1. Realistically not able to counter enemy attack, the counter window is very shallow.
2. Very fast pace, customizable das/arr and no line delay creates extremely high damage spikes. 100apm is not a simple 1.6 lines per second.
3. Garbage lines are very forgiving; it's very common to have something like 12 depth well (triple quad), multiplying overall APM even more. 
4. Practically 'guaranteed kill' type of damage exists, being significantly more common as player skill goes up. Also, because of 1), 2), 3), meta is to keep playfield low, and this mostly happens from RNG(garbage line) during that process, instead of a player's setup.


I'm pretty sure you're with me on how you feel about 4-wide based on what you did to combo damage. **To put it in to perspective, frustration-wise it's sometimes worse than 4-wide.** Yes 4-wide fiesta is gone, nothing is a dominating strategy, i love it, but now it's more like whatever you do, your death cause is from uncounterable, guaranteed death combo which is basically planned by a RNG generator. 

**Yes even firestorm will probably die from a SS player's DT cannon into double tetris because a RNG generator decided to generate 8 deep hole on leftmost column, and it just did 4+7+5+6 = 22 damage in 0.5 seconds which he realistically cannot counter.** See what's happening here?

Guideline tetris games are also RNG, but no, generating few lucky garbage line doesn't **instantly** kill you, and killer attack relies significantly less on the RNG. 

**At least (the pre-nerf) 4-wide took time to execute;** thus equal amount of time for enemy to **respond**. 
It's 40 blocks over 7.0 seconds assuming 13 combo 2pps. I know this usually results in death, but at least you can try to work against it.

current tetrio's damage spike as in '20 blocks in 0.5 seconds' is a guaranteed death, it's definitely frustrating by itself, and it's worse when you know that it's outcome of a RNG. 

I really don't care too much what you do to fix the issue, but once again I would identify this as a problem, and I feel like something has to be done.

**tl:dr, basically, high rank plays are hilarious. help.**

**Comment by Ricardito10** [comment]
I dont want to add too much to this, but perhaps a change of the RNG of garbage is all thats needed to rectify the issue, if Im recalling correctly. Because if for example, you are sent an 8 spike, 9 times out of 10 all of those 8 lines are in a well, ready to be used to spike the opponent out of nowhere.

**Comment by csq-curiosity** [comment]
So, I want to go over the main thread point-by-point.
Issues raised:
> 1. Players are forced to play in a very downstack-heavy playstyle, especially on high(SS+) level gameplay.

This is indeed true.

> 2. Game-finishing attacks, aka a Quad/T-spin combo, is mostly a result of RNG, and is less related to skill.
> This issue is closely tied to (1)'s issue. Because you realistically cannot negate incoming damage promptly, you're forced to maintain a low playfield to not lose in a frustrating way. This discourages "setting up" for your back-to-backs as complicated tspins require higher playfield.

This one is a little interesting, because the response may vary depending on whether you are capable of "opponent assesment"; that is, whether you can check the opponent's board mid-play and react accordingly depending on the situation. I'll admit that I myself is quite terrible at this, but those who are adept at it will have various responses:
- If the opponent has a major spike or a potentially finishing blow, they will wait for the incoming attack and cancel quad/tsd with quad/tsd.
- If the opponent does not have an immediate threat set up, or otherwise struggling from misdrops and/or high stacks, they will opt to set up complex t-spin setups for a finishing blow.
- Sometimes they will intentionally not cancel garbage and induce crossfire, so as to cause immediate trouble to the opponent. This frequently happens when both players go for an all clear opening game.

> This leads to players not being able to setup high damage attacks by themselves, instead rely on garbage line which seems to have very forgiving algorithm, making Quad very easy. However combing into Quad is another issue, as it requires I piece at the right time, when you have the right playfield.

I'd argue quite the opposite. I've seen players *very frequently* capitalizing on garbage and chain them quite easily, with simple TSD setups using the center well, to a side well donation, and sometimes STSD or fractal setups. The thing is, most high rank players just opt for holding a T or an I piece in order to deliver attacks whenever needed. Granted, this idea along with the current garbage system makes T or I misdrops very punishing without proper downstacking or misdrop management in general, but looking at players having already adapted to this system, I don't see a clear reason as to why change. Indeed they should be addressed if there happens to be an absolutely broken issue or set of issues without any clear ways to get around it, but that simply doesn't seem to be the case.

Suggested solution:
> Add a time based window to counter incoming garbage lines. Time i suggest is 2-3 seconds, preferbly 3s, as It allows players playing in 2PPS to cycle through one bag with hold's help.

Now I am in favor of this solution provided that it is an optional choice in custom rooms, but I don't think it should be a staple of this game as it basically removes a design feature tetrio has to offer.

Also, 2~3 seconds seems ridiculously long, as it gives the opponent too much time to downstack out of a troublesome situation. Consider that jstris only has a 500ms (0.5 seconds) window of time before the garbage pushes up to your stack. I get that 0.5 seconds is barely enough time for you to cycle 1 bag in 2 pps, but that is because it is not meant to be. If you have a vision fast enough to play this game in 2 pps, then a 0.5 second window is enough time to get yourself alerted that you are in trouble and you should react to it, whether you tank the garbage or counterattack. I feel like any window longer than 0.5 seconds either stops you from making tactical decisions fast enough to respond to the opponent, or gives you too much time to react to the point that games drag on for so long.

The default garbage travel time in tetrio is 20 frames, which is about 0.333... seconds. It also works in a slightly different way than most games; while jstris's 0.5 second window both shows you the garbage and gives you the chance to cancel it, tetrio's 0.333 second window shows you that the garbage is incoming but you cannot cancel the garbage until it actually hits your board. While I do think this is serviceable although a bit fast, one problem with the game is that **the game seems to *skip* the 0.333 second window of time when it finds that you are lagging behind.** This is possibly one root of the problem when you are arguing that you don't have enough time to react to the garbage. I would suggest not to skip this window at all times.

Before I get into the conclusion, let me make myself clear in that **I *am* very biased to the status quo as I find myself mostly accustomed to the garbage mechanic of this game**. Not everyone will be satisfied with my claims, so feel free to comment on any point you disagree on, or have opposing clues/evidences.

Conclusion: It seems like a lot of the upper echelon players have already developed various strategies that are effective in the tetrio multiplayer environment. Downstacking, spiking, B2B abuse, (yes, some players can and do maintain B2B despite the intense spikes the games tend to offer) etc. Players have also found ways to spike (a simple example being a TSD on top of a quad well) and dealing with incoming garbage spikes. (an example being waiting for the garbage and partially cancelling, and using the rest to your advantage) So it seems like the downstack-heavy gameplay is not inherently an issue and simply becoming a meta, a strategy unique to the experience of tetrio. There definitely are players enjoying this mechanic, and I think it is should stay that way.

Below this are the minor nitpicks:

> Realistically not able to counter enemy attack, the counter window is very shallow.

You can just wait out until the attack actually arrives on the board and cancel them, or better yet, watch the opponent's board for an incoming attack. Now I get that the latter doesn't always work when either you or the opponent is lagging, but it's still an option you can take.

> Garbage lines are very forgiving; it's very common to have something like 12 depth well (triple quad), multiplying overall APM even more.

It's actually surprisingly uncommon to have a 12 depth well unless you get very lucky. Getting a 12 depth well means that either you got a 12 line spike from an all clear or something, you got multiple waves of attack that all happened to land on a single column, or you had your own stack complemented by more rows of garbage. It is even more uncommon because the garbage is always broken into 8 lines, and getting 8 lines instantly will likely prompt a downstack response, so all of the above hypothetical cases have the condition that you don't downstack the incoming garbage. Having a 12+ line stack in your side in general is a risky move, so I'd argue that a 12 depth well that you've gotten is a reward for your troubles. It is also very uncommon to strike a triple quad at once because that itself is quite RNG.

> Practically 'guaranteed kill' type of damage exists, being significantly more common as player skill goes up. Also, because of 1), 2), 3), meta is to keep playfield low, and this 'kill' mostly happens during that process (downstacking 14 damage quad etc), and it's more of a result from RNG, not a player's setup.

This remark looks to me that you are underestimating how profoundly difficult a skill downstack is; while it is undeniable that downstacking with certain piece combinations in certain situations is messy at best, you have a surprisingly major amount of control on how you can downstack out of a bad situation. The order of pieces, the orientation of each piece, opting to slightly make your stack messy to open up the well below you, and even setting up a downstack finish to a TSD. There are so many different ways you can approach downstacking, and putting this all into a result from RNG simply lacks taste to me.

> Yes even firestorm will probably die from a SS player's DT cannon into double quad because a RNG generator decided to generate 8 deep hole on leftmost column, and it just did 4+7+5+6 = 22 damage in 0.5 seconds which he realistically cannot counter. See what's happening here?

Doing 22 damage in 0.5 seconds not only requires you to play in solid 4~5 pps which is hardly ever conceivable in X tier let alone SS, it also requires the pieces to appear in T - T - I - I in that strict order, with one of the pieces give or take with the hold bag. That is quite an impossibly absurd luck. Considering the probabilistic expectations of pieces, the damage you have outlined is more conceivable in around 6 seconds. Doing 22 damage in 6 seconds translates to 220 APM, which is hefty, but APM bursts of that level is not very uncommon for X tiers like Firestorm, especially in the context of early games. It is not inconceivable that they can survive the onslaught.

There are also multiple confounding factors, two most immediate being that Firestorm likely won't wait out 6 seconds doing nothing, and he will likely stand a chance since the garbage lines are broken down into waves of at most 8 lines, giving him 2 more chances to counter before completely topping out. It is also important to point out that DT is a fairly big opener involving 13 pieces to set up, which is going to take time in its own. As a final insult to injury, Firestorm's Blitz record suggests that he can do 2 all clears within around 6~7 seconds. All clears do a minimum of 11 lines of damage, so 2 all clears completely negate the 22 line onslaught.

A 22 spike within 0.5 seconds is an incomprehensible level of garbage even for X tiers; nobody will survive that attack, but nobody will be able to deliver that amount of attack in the first place. Just to show you how absurd that is, a 22 spike within 0.5 seconds translates to 2640 APM, and this is more than 5 times higher than the recorded peak APM tetrio has ever seen, which is 455 APM. (https://youtu.be/e0vPsEQgQfU) Even an AI operating in 13.5 pps cannot reach a third of that APM! (https://youtu.be/rZWuesSBZHk)

**Comment by Ricardito10** [comment]
Yknow, this is making me think whether I should hold off with Tetra league until the whole damage table is locked in. I DM'd osk the other day, and they didn't say much as to whether anything is going to be changed or not, though with how they responded, I have a feeling this won't be changing anytime soon. Either way, thats my second cent.

And after rereading the post above, I'd side with csq-curiosity with how the meta is right now. I've actually used a combo spike a couple of times and it resulted in getting the win for me, so... yeah.

**Comment by bmjeon5957** [comment]
> This remark looks to me that you are underestimating how profoundly difficult a skill downstack is; while it is undeniable that downstacking with certain piece combinations in certain situations is messy at best, you have a surprisingly major amount of control on how you can downstack out of a bad situation. The order of pieces, the orientation of each piece, opting to slightly make your stack messy to open up the well below you, and even setting up a downstack finish to a TSD. There are so many different ways you can approach downstacking, and putting this all into a result from RNG simply lacks taste to me.

This is the main issue, and **No, I'm not underestimating the ability to downstack. If not, I'm overestimating the people's ability to downstack**, because I'm assuming people would basically always 10 combo downstack when they can. Problem here is that including a Quad or T-spin is significantly more of an RNG. and a 10 combo Quad is basically almost a guaranteed kill. 

I understand how much skill you need to succeed in order to downstack, let's put it this way, but pro poker players having extremely high likelyhood of stomping me doesn't really make poker a *skill based game*. I mean it is, but it isn't. 

Recall that I've specifically mentioned "high level play"?  **It really doesn't matter if you're basically a cold clear or not,** like cold clear would very often find its way to downstack to 10 combo (obviously because it's genius AI), **I have very rarely seen even cold clear having a Quad or Tspin at high combo, and when it happens once in a very while, it is usually result of an RNG**.


> I'd argue quite the opposite. I've seen players very frequently capitalizing on garbage and chain them quite easily, with simple TSD setups using the center well, to a side well donation, and sometimes STSD or fractal setups. The thing is, most high rank players just opt for holding a T or an I piece in order to deliver attacks whenever needed. Granted, this idea along with the current garbage system makes T or I misdrops very punishing without proper downstacking or misdrop management in general, but looking at players having already adapted to this system, I don't see a clear reason as to why change. Indeed they should be addressed if there happens to be an absolutely broken issue or set of issues without any clear ways to get around it, but that simply doesn't seem to be the case.

People stacking according to garbage holes is clearly a thing, and it always have been. That's not exactly my point. 

The issue is that my "killing blow" is both combination of 1) first of all, need to be able to combo-downstack 2) 4 deep hole at the right place  3) and right sequence of pieces that would leave my Quad well open/ends with an I. Once again while there definitely is skill involved there, however it's also a skill that needs a lot of help from RNG. 

Friendly reminder that I'm assuming both players are fairly high in skills who will very much keep their field close to ideally, and would downstack almost ideally. and THEN things become pretty RNG heavy, and that RNG Quad combo doing 14 damage in 0 second would feel really lame for opposing player.

> I feel like any window longer than 0.5 seconds either stops you from making tactical decisions fast enough to respond to the opponent, or gives you too much time to react to the point that games drag on for so long.

Everything you mentioned about "2 seconds too long, 0.5 maybe enough" Is something i can respect actually, I actually don't mind as long as it becomes a system where you can realistically counter enemy attack.


> So it seems like the downstack-heavy gameplay is not inherently an issue and simply becoming a meta, a strategy unique to the experience of tetrio. 

Which I do identify it as an issue because downstacking into Quad IS significantly more RNG compared to regular downstacking, as once again, there are basically three randomness involved in this compared to regular downstacking's one. I mean, sure, people play poker, but I'd prefer less RNGs in tetrimino stacking game.



> You can just wait out until the attack actually arrives on the board and cancel them, or better yet, watch the opponent's board for an incoming attack. Now I get that the latter doesn't always work when either you or the opponent is lagging, but it's still an option you can take.

And just to add stuff, I'd have to say this is basically always, I play from Korea, and I've seen many players who has this exact same issue, especially in my country. It seems likely that who doesn't play close to tetr.io server suffers same issue. This might be one of the biggest factors that makes player feel different, but network is network, I guess.




> All firestorm stuff

I'm skipping this as 1) Yes i exaggerated 2) My point is still made 3) You likely have mentioned nitpick because of these and would probably be unhealthy to drag this topic

**Comment by csq-curiosity** [comment]
Reading this comment, it seems clear that I was missing the mark of the point you were trying to make. Sorry about that.

AFAIK the current garbage system greatly rewarding downstacking is sort of a "compensation", as trying to combo down in guideline games usually devolves into an endless spire of cheese race. This still happens in tetrio even with the "enhanced" garbage system. What I think the devs envisioned for the game is by rewarding downstacking in the form of big spikes, the player can hopefully escape the spire more easily and turn the tables. And it works well, enabling for a more dynamic play in the form of huge comebacks.

What makes the downstack spikes powerful (or _too_ powerful as you would suggest) relies on two factors:
1. The garbage system outlined above, which, the line of reasoning itself sounds nice to me
2. Garbage being always clean

A similar type of this issue was raised back in April when a further nerf for 4-wide or similar combo tech was raised in #262. User alex-ong remarked that a 4-wide into a quad was a killer move, which doesn't seem to be too different from what you mean by a (big number) quad is a guaranteed kill. Fixing this butchers the downstack back to the guideline standards, which revives the old problem of trying to downstack out of a sticky situation being a spire.

In that thread, osk maintained the position of keeping the garbage system as is, as I quote:

> 4wide is very harshly nerfed already, I feel any further nerfs would make downstacking unviable as well and/or combos entirely.

Now, if you disagree with this sentiment, I'd suggest going to that thread and continue the discussion there.

A couple of more nitpicks before I make my major point:

> The issue is that my "killing blow" is both combination of 1) first of all, need to be able to combo-downstack 2) 4 deep hole at the right place 3) and right sequence of pieces that would leave my Quad well open/ends with an I.

Recall that I mentioned above a 4-wide into a quad being a killer move. This means that while 2) is likely necessary, 1) and 3) can be replaced with stacking a 2~4wide around the garbage hole to combo into a quad that way. If you were to point out that this is a cheesy move I totally agree, and I may have unintentionally done this a couple of times :P 

> Friendly reminder that I'm assuming both players are fairly high in skills who will very much keep their field close to ideally, and would downstack almost ideally. and THEN things become pretty RNG heavy, and that RNG Quad combo doing 14 damage in 0 second would feel really lame for opposing player.

As someone who believes to have been in this situation a lot, the matches usually pan out as B2B crossfire -> downstack -> rinse and repeat. In the downstack phase there are downstack spikes, but a lot of them tend to be 8 spikes at most, which is survivable a lot of the times. Also, if a 10+ spike is indeed looming, you get a sort of a "warning signal" in the form of 1~2 garbage lines continuously flowing towards you, giving a bit of time to prepare yourself. (usually by taking T-spin setups) Getting a 14 spike out of the blue (and by that I mean suddenly, with literally no time to prepare) is very rare unless you get slammed with a quad PC or a TSD+quad chain from the opponent who has built up 8+ B2B.

Okay, now onto my major point.

> And just to add stuff, I'd have to say this is basically always, I play from Korea, and I've seen many players who has this exact same issue, especially in my country. It seems likely that who doesn't play close to tetr.io server suffers same issue. This might be one of the biggest factors that makes player feel different, but network is network, I guess.

I'm guessing that "this" refers to

> Now I get that the latter doesn't always work when either you or the opponent is lagging

And as another Korean player myself I can totally relate. Which is why in my first comment I mentioned

> one problem with the game is that the game seems to skip the 0.333 second window of time when it finds that you are lagging behind. This is possibly one root of the problem when you are arguing that you don't have enough time to react to the garbage. I would suggest not to skip this window at all times.

Another way of approaching, on top of that suggestion, is to **make the garbage travel speed _variable_ based on its size**. So while small amounts of garbage, like 1 or 2 lines, can remain on the 0.333 second window, whereas big spikes, such as 11 lines from a PC or a quad finish from a huge downstack combo, can travel as slow as 1 second. Osk has already made the option of changing garbage travel speed available, so this doesn't seem to be completely nonviable. Now, due to the nature of garbage system this game has, you _cannot_ cancel garbage while it's travelling towards you, but it can alert you enough to prepare for a counterattack to cancel them.

**Comment by bmjeon5957** [comment]
> In that thread, osk maintained the position of keeping the garbage system as is, as I quote:
> 
> 4wide is very harshly nerfed already, I feel any further nerfs would make downstacking unviable as well and/or combos entirely.
> 
> Now, if you disagree with this sentiment, I'd suggest going to that thread and continue the discussion there.


Ok i can probably understand the reason why my opinion on "combing" is some interpreted as *too strong*, but it's not. I'm actually one of the people who thinks 4 wide/combo was nerfed *too much*, but I hope quoting this helps  : 

> All this is quite closely tied to ability to counter enemy attack, since as an example, 20 damage over 0 seconds is something that literally nobody can deal with, while 30 damage over 2 seconds is I mean, likely a death, but is something you can deal with and would significantly reduce the problem of "RNG finishes the game".

It's another topic so I won't really dig deeper, but basically, I'm trying to say that instead of getting 14 damage over 0 seconds (which you did explain later on your post, will talk about it again) I'd take 20 damage over 2 seconds, for example. 

My whole thing with "need fixed amount of time window to counter(aka, being able to counter garbage while traveling)" is heavily related to this though for sure; it makes 14 damage over 0 seconds to 14 damage over 1 second, which would make feel players (or me at least) more like "shit i couldn't prepare, my bad" instead of "well that's something i cannot do anything about". 

I did mention "quad combo needs nerf especially if we don't get to counter garbage while travelling", However, I wasn't exactly trying to say the amount of bonus it gets is too huge, was trying to say that it's combination of "not able to counter" and "it's a spike over 0 second".

Honestly I'd still want to say that being able to counter garbage during its travel time, is the best solution to fix all the problem at once, but if anyone else does think that this is tetr.io's unique trait and think it's worth saying, I mean fine, but at least

Another way of approaching, on top of that suggestion, is to make the garbage travel speed variable based on its size. So while small amounts of garbage, like 1 or 2 lines, can remain on the 0.333 second window, whereas big spikes, such as 11 lines from a PC or a quad finish from a huge downstack combo, can travel as slow as 1 second.

+1. This does sound like an amazing idea regardless, and combined with "don't skip travel time on lag" actually might potentially fix *my* issue too with the spikes even being able to counter during garbage travel time.

**Comment by Manabender** [comment]
My two cents, as a player currently hovering around 17-18k TR:

I do think a nerf to the combo bonus on quads is in order. Almost no matter how they are achieved, if you quad on a 3 or higher combo, the damage is ridiculous and undeserved. There doesn't seem to be a known way to reliably perform them. When you do, it's because you got a lucky bag. Luck alone should not make victory.

That being said, I do think it's OK to "get lucky" occasionally. That's the entire reason a match consists of multiple games. Sometimes, you just get lucky. If it's rare enough, that's fine. But in a first-to-3 match, the more skilled player is likely to win even if the other player gets a lucky game. The nature of high-level play is also why matches go to FT5 and FT7 at high ranks, because those lucky spikes become a bit more common.

**Comment by bmjeon5957** [comment]
Once again, It's function of both RNGness and frustration from a sudden death that is significantly more common in tetrio, and I'm really not exaggerating on that people are complaining this issue on a different website and more importantly, silently leave.

Despite this opinion has gotten very significant positive reaction like a month ago, It's rather getting *controversial* now here in github. As time goes on, the majority of opinion who has **stayed** in tetr.io would be biased towards "keep what was changed in tetr.io" for sure, because I mean, **it's pretty clear that people who can bare these frustrations would stay**. I shit you not, It's too late when most of people who were initially interested in, leaves due to the issues.

**Tetr.io has huge potential and after some problems solved, Tetr.io deserves to be the best PvP platform on all tetrimino stacker kind. Now this being hindered by some "give it a try" feature that would potentially make people leave, is just really depressing and sad in my opinion.** 

While it might be interesting to some people, this *new feature* wouldn't really drag in any new playerbase. If it's a give and take, like OK sure, **but it's literally only acting as a factor that splits the playerbase.

I'm very certain that not many would say "tetr.io doesn't have a time window to cancel out garbage, and that is why i play tetr.io"**

I'm repeatedly mentioning this problem because l'm quite frequently seeing people mentioning tetr.io has really weirdly spiky  nature that they do not enjoy, and sometimes they think it's entirely function of lag, concluding it as "tetr.io lags, it sucks" and obviously quit. I mean their conclusions were wrong, but yes tetr.io is *extra* spikey because (read first post), which it didn't need to be.

> The nature of high-level play is also why matches go to FT5 and FT7 at high ranks, because those lucky spikes become a bit more common.


The fact that higher FT somewhat *mitigating the RNG feature* doesn't exactly reduce the total frustration from both RNGness and a sudden death that realistically cannot be dealt with. I play FT7 myself and maybe only half of my games feel like they're deserving.

 Just as an example, Would you rather play a FT1 match where every game would be a GG, or a FT7 game where like 80% of the games would either feel like RNG and/or undeserved and only 20% of the games are actually interesting? I mean, I'd pick a GG FT1 100%.

Yes putting tetra league at FT7 helped X/U game's **result** being less RNG. But no, it did not reduce total frustration generated by RNG and/or undeserving death. and yes people leave the game for it. 



Time to think it the other way. "Is (basically) not having counter window really a huge issue and needs to be changed to have counter window immediately?" I mean I'd say yeah, but you're free to say no. I'd say it's controversial, sure.

but "Did we really need to remove the counter window? Was it causing any problem? did people leave because of that and will people leave because of it?" and answer seems like a very obvious no.

The opposite question though, "Did people leave because of no counter window?" seems like a yes to me. It might just be the community I'm in, but I feel like the percentage is quite high. Literally None of you disagreed that we're playing more RNG and more Spiky game compared to the already existing ones.

**Comment by o5k** [comment]
> Another way of approaching, on top of that suggestion, is to **make the garbage travel speed variable based on its size**. So while small amounts of garbage, like 1 or 2 lines, can remain on the 0.333 second window, whereas big spikes, such as 11 lines from a PC or a quad finish from a huge downstack combo, can travel as slow as 1 second. Osk has already made the option of changing garbage travel speed available, so this doesn't seem to be completely nonviable. Now, due to the nature of garbage system this game has, you cannot cancel garbage while it's travelling towards you, but it can alert you enough to prepare for a counterattack to cancel them.

I have the feeling this would actually cause bigger issues, as the garbage cannot be canceled during that period. In that sense, all it might do is cause more confusion and make it harder to time some attacks (as you can't react to the sound of an attack coming in anymore, you have to gauge how big the attack is and then adjust your timing to that). A potential fix would be making the timing based on the sound played, so a "large attack" sound will always travel X milliseconds. It will help you either way if your reaction speed is above 333ms though, which may for many very well be the case.

By the way, "skipping the travel time on lag" shouldn't be happening. If it does, please get me a video of it or something! It should _always_ be 20 ingame frames by default.

(By the way, please keep it calm and collected here. This is a place for discussion, not shouting matches. Thank you!)

**Comment by Xelitik** [comment]
I think a time based window to place pieces after receiving attacks would be nice, especially in FFAs, when some players cannot screen watch in higher rank lobbies because of a low-end PC. 
Nothing is more frustrating then taking in so much garbage unexpectedly from multiple people in less then a second and dying because you weren't prepared with a defense the moment the garbage came. Of course it would go away progressively until it no longer exists with something like margin time.

Just a chance to play without dying 30 seconds in would be nice in FFAs. Especially with all the waiting involved in joining on going games, waiting for others to finish after you top out, victory screens, waiting for a host to start the game.

This just doesn't feel fair in my eyes (sorry for background noise, was in a discord call): https://youtu.be/5CVUIC4E768

**Comment by Ricardito10** [comment]
I feel like this issue is gonna cause a massive breakdown in the Tetr.io community.

**Comment by bmjeon5957** [comment]
It's really good to see what csq suggest being added, now can we also get a selectable feature to actually counter garbage while traveling? feels like it's good time to test that also

**Comment by xenonsb** [comment]
23k TR playing here (quick play at about 2.4 pps/80 apm), just my 2 cents

After the new spike delay change, quick play's become quite weird.. no longer can I time the garbage reversals right, and it's adding a bit of brain lag moving forward.

What used to work as strategic spikes with fractals/STSD causing b2b spikes were no longer ko-ing people - instead I'm getting returned some nasty cheesed garbage. The other down-side I'm seeing is my over-inflated APM, seeing about a 20-30 additional APM on average per game. The upside though - I am able to escape out of some severe spikes by carefully downstacking, for those edge games that I get about 20 lines 10 seconds into the game.

Also, this delay favours short combos to send immediate garbage, whereas larger spikes can be mitigated by careful downstacking. Not overly fond of that 👎

**Comment by Ricardito10** [comment]
Nothing happened here in a while, hows the situation looking?

**Comment by bmjeon5957** [comment]
I've been stop playing for few weeks now for obvious reason(s), kind of came back from email notification :P

Skimming through, I've also seen another very similar suggestion ( https://discordapp.com/channels/673303546107658242/673305735811170305/742202491608760370 ) in #feature-request, which also got about 23 checks. This conversation apparently also involved osk? I'm not sure..

I get that some people are fine with how it is now, but I very rarely see people who suggests that that we should NOT change. Somehow at the same time, major suggestions to go back. 

 I seldom participate in regular chats so I have no idea how people complains or thinks besides some "memes" people post about RNG, but as Caboozled_Pie suggests on his suggestion :

>  it's worth noting that many people who have put in large amounts of effort into the game have failed to adjust, and frankly, are unable to properly appreciate the game.

which actually made people, including me, go back to the other stacker game, but as much as I sincerely think maybe it isn't a big deal for the game itself,

> Nonetheless, many people in the community have been split over this, so I personally consider it worth having an honest and direct discussion about what should be done. 

I mean that's what he's done, and he's done it greatly, but I believe this issue deserves a proper, official discussion, as the one you did back in with the great-180-rotation that probably should result one way or another.

**Comment by ZaptorZap** [comment]
Heh, I guess this got fixed with [6.1.2](https://tetr.io/about/patchnotes/#chlog_6_1_2). The chosen solution here was solution (1), although the delay still holds at above [20 frames(@60fps, this is nearly 333 milliseconds)](https://tetris.wiki/TETR.IO#Garbage_Travel_Speed:_.22Passthrough.22). Seemed like you were looking for a **lot** longer of a delay, but this issue kinda entirely lost steam for the _years_ it was open anyway, so I'm closing this as completed.
